spark-partition-sizing/src/test/scala/za/co/absa/spark/partition/sizing/sizer/FromDataframeSampleSizerTest.scala

```diff
@@ -34,13 +37,18 @@ class FromDataframeSampleSizerTest extends AnyFunSuite with DummyDatasets {
   }
 
   test("test deeper nested dataframe") {
-    val inputDf = spark.read
+    def readInput(): DataFrame = spark.read
       .schema(testCaseSchema)
       .json(relativeToResourcePath(nestedFilePath))
 
-    assert(sizer.performRowSizing(inputDf) < 3000)
-    //the number of samples should be higher than 0
-    assert(sizer.performRowSizing(inputDf) > 0)
+    eventually(timeout(scaled(1.seconds)), interval(scaled(500.millis))) {
+      val inputDf = readInput()
+      assert(!inputDf.isEmpty, "input should not be empty") // eventually will retry on this if empty
+
+      assert(sizer.performRowSizing(inputDf) < 3000)
+      //the number of samples should be higher than 0
+      assert(sizer.performRowSizing(inputDf) > 0)
```

Comment:
Cool solution, excellent finding.
I would keep only the reading and non-empty testing in the eventually. The later, actual test asserts I would keep outside (lines 48-50).
Would also consider making the reading of the resource file into `DataFrame` a method of `DummyDataset`